logserver/src/test/java/com/yahoo/logserver/handlers/archive/FilesArchivedTestCase.java

```diff
@@ -0,0 +1,126 @@
+
+    private void makeLogfile(File dir, String name, long hours) throws IOException {
+        File f = new File(dir, name);
+        f.getParentFile().mkdirs();
+        new FileWriter(f).write("foo bar baz\n");
+        long now = System.currentTimeMillis();
+        f.setLastModified(now - (hours * 3600 * 1000));
+    }
+
+    void checkExist(File dir, String name) {
+        assertTrue(new File(dir, name).isFile());
+    }
+    void checkNoExist(File dir, String name) {
+        assertFalse(new File(dir, name).isFile());
+    }
+
+    @Test
+    public void testMaintenance() throws java.io.IOException {
+        File tmpDir = temporaryFolder.newFolder();
+        try {
+            makeLogfile(tmpDir, "2018/11/20/13-0", 35*24);
+            makeLogfile(tmpDir, "2018/11/21/13-0", 34*24);
+            makeLogfile(tmpDir, "2018/12/28/13-0", 3*24);
+            makeLogfile(tmpDir, "2018/12/29/13-0", 2*24);
+            makeLogfile(tmpDir, "2018/12/30/13-0", 1*24);
+            makeLogfile(tmpDir, "2018/12/31/14-0", 3);
+            makeLogfile(tmpDir, "2018/12/31/16-0", 1);
+            makeLogfile(tmpDir, "2018/12/31/17-0", 0);
+            dumpFiles(tmpDir, "before archive maintenance");
+            FilesArchived a = new FilesArchived(tmpDir);
+            dumpFiles(tmpDir, "after archive maintenance");
+            checkExist(tmpDir, "2018/12/31/17-0");
+            checkExist(tmpDir, "2018/12/31/16-0");
+            checkExist(tmpDir, "2018/12/31/14-0.gz");
+            checkExist(tmpDir, "2018/12/28/13-0.gz");
+            checkExist(tmpDir, "2018/12/29/13-0.gz");
+            checkExist(tmpDir, "2018/12/30/13-0.gz");
+
+            checkNoExist(tmpDir, "2018/12/31/17-0.gz");
+            checkNoExist(tmpDir, "2018/12/31/16-0.gz");
+            checkNoExist(tmpDir, "2018/12/31/14-0");
+            checkNoExist(tmpDir, "2018/12/28/13-0");
+            checkNoExist(tmpDir, "2018/12/29/13-0");
+            checkNoExist(tmpDir, "2018/12/30/13-0");
+
+            checkNoExist(tmpDir, "2018/11/20/13-0");
+            checkNoExist(tmpDir, "2018/11/20/13-0.gz");
+            checkNoExist(tmpDir, "2018/11/21/13-0");
+            checkNoExist(tmpDir, "2018/11/21/13-0.gz");
+
+            makeLogfile(tmpDir, "2018/12/31/16-0", 3);
+            makeLogfile(tmpDir, "2018/12/31/17-0", 3);
+            makeLogfile(tmpDir, "2018/12/31/17-1", 1);
+            makeLogfile(tmpDir, "2018/12/31/17-2", 0);
+
+            dumpFiles(tmpDir, "before second archive maintenance");
+            a.maintenance();
+            dumpFiles(tmpDir, "after second archive maintenance");
+
+            checkExist(tmpDir, "2018/12/31/17-2");
+            checkExist(tmpDir, "2018/12/31/17-1");
+            checkExist(tmpDir, "2018/12/31/16-0.gz");
+            checkExist(tmpDir, "2018/12/31/17-0.gz");
+
+            checkNoExist(tmpDir, "2018/12/31/16-0");
+            checkNoExist(tmpDir, "2018/12/31/17-0");
+        } finally {
+            IOUtils.recursiveDeleteDir(tmpDir);
```

Comment:
Btw, this is not necessary as `TemporaryFolder` will delete the directory once the test methods is finishes (even if test fails).